stl_to_tscn.py

```diff
@@ -0,0 +1,245 @@
+                    elif jpg == "normal":
+                        sub_resource += Textures.normal_options + "normal_texture = ExtResource( " + str(ext_resource_id) + " )\n"
+                    elif jpg == "depth":
+                        sub_resource += Textures.depth_options + "depth_texture = ExtResource( " + str(ext_resource_id) + " )\n"
+
+                    # don't forget to increment ext_resource_id for the next one
+                    ext_resource_id += 1
+
+                # add uv1_scale if texture has it
+                if Textures.texture_dict[texture].uv1_scale:
+                    sub_resource += "uv1_scale = " + Textures.texture_dict[texture].uv1_scale + "\n"
+
+                # add the sub resource's index to texture_index so we can use it later 
+                sub_resource += "\n"
+                texture_index[texture] = sub_resource_id
+                sub_resource_id += 1 
+
+            elif texture in Textures.other_textures:
+                # only need sub_resource
+                sub_resource += "[sub_resource type=\"SpatialMaterial\" id=" + str(sub_resource_id) + "]\n" + Textures.other_textures[texture]
+                
+                # add the sub resource's index to texture_index so we can use it later 
+                sub_resource += "\n"
+                texture_index[texture] = sub_resource_id
+                sub_resource_id += 1 
+
+
+        #####################
+        # generate obj mesh #
+        #####################
+        stl_file = m["stl_file"]
+
+        # output filename without a file extension
+        output_prefix = output_folder + "/" + stl_file[:-4]
+        
+        # Generate .obj file
+        generate_obj_file(stl_file, output_prefix, m["uv_map"])
+        # add it to ext_resource
+        ext_resource += "[ext_resource path=\"res://models/" + output_prefix + \
+                        ".obj\" type=\"ArrayMesh\" id=" + str(ext_resource_id) + "]\n"
+        # add it to nodes
+        nodes += "[node name=\"" + stl_file[:-4] + "\" type=\"MeshInstance\" parent=\".\"]\nmesh = ExtResource( " \
+                    + str(ext_resource_id) + " )\nmaterial/0 = SubResource( " + str(texture_index[texture]) + " )\n\n"
+
+        # move on to next ext_resource
+        ext_resource_id = ext_resource_id + 1
+
+        #######################
+        # Generate collisions #
+        #######################
+        if m["collisions"]:
+            generate_collisons(stl_file, output_prefix)
+            # add it to ext_resource
+            ext_resource += "[ext_resource path=\"res://models/" + output_prefix + \
+                            ".glb\" type=\"PackedScene\" id=" + str(ext_resource_id) + "]\n"
+                        
+            nodes += "[node name=\"" + stl_file[:-4] + "Col\" parent=\".\" instance=ExtResource( " + str(ext_resource_id) + " )]\n\n"
+
+            # move on to next ext_resource
+            ext_resource_id = ext_resource_id + 1
```

Comment:
functionalize.

Reply:
`generate_collisions` is already a function that generates the `.glb` collision files, and it would be awkward to put the other lines here into the function because that would require passing in the `ext_resource` and `nodes` string into the function.